apps/condo/domains/acquiring/schema/RegisterMultiPaymentForVirtualReceiptService.js

```diff
@@ -0,0 +1,223 @@
+const {
+    getAcquiringIntegrationContextFormula,
+    FeeDistribution,
+} = require('@condo/domains/acquiring/utils/serverSchema/feeDistribution')
+const { GQLError, GQLErrorCode: { BAD_USER_INPUT } } = require('@condo/keystone/errors')
+const { DV_VERSION_MISMATCH, WRONG_FORMAT } = require('@condo/domains/common/constants/errors')
+const { checkDvAndSender } = require('@condo/keystone/plugins/dvAndSender')
+
+const {
+    RECEIPTS_HAVE_NEGATIVE_TO_PAY_VALUE,
+    RECEIPT_HAVE_INVALID_TO_PAY_VALUE,
+    ACQUIRING_INTEGRATION_IS_DELETED,
+    ACQUIRING_INTEGRATION_CONTEXT_IS_DELETED,
+} = require('../constants/errors')
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const errors = {
+    DV_VERSION_MISMATCH: {
+        mutation: 'registerMultiPaymentForVirtualReceipt',
+        variable: ['data', 'dv'],
+        code: BAD_USER_INPUT,
+        type: DV_VERSION_MISMATCH,
+        message: 'Wrong value for data version number',
+    },
+    WRONG_SENDER_FORMAT: {
+        mutation: 'registerMultiPaymentForVirtualReceipt',
+        variable: ['data', 'sender'],
+        code: BAD_USER_INPUT,
+        type: WRONG_FORMAT,
+        message: 'Invalid format of "sender" field value. {details}',
+        correctExample: '{ dv: 1, fingerprint: \'example-fingerprint-alphanumeric-value\'}',
+    },
+    ACQUIRING_INTEGRATION_CONTEXT_IS_DELETED: {
+        mutation: 'registerMultiPaymentForVirtualReceipt',
+        variable: ['data', 'acquiringIntegrationContext', 'id'],
+        code: BAD_USER_INPUT,
+        type: ACQUIRING_INTEGRATION_CONTEXT_IS_DELETED,
+        message: 'Cannot pay via deleted acquiring integration context',
+    },
+    ACQUIRING_INTEGRATION_IS_DELETED: {
+        mutation: 'registerMultiPaymentForVirtualReceipt',
+        variable: ['data', 'acquiringIntegrationContext', 'id'],
+        code: BAD_USER_INPUT,
+        type: ACQUIRING_INTEGRATION_IS_DELETED,
+        message: 'Cannot pay via deleted acquiring integration with id "{id}"',
+    },
+    RECEIPT_HAVE_INVALID_TO_PAY_VALUE: {
+        mutation: 'registerMultiPaymentForVirtualReceipt',
+        code: BAD_USER_INPUT,
+        type: RECEIPT_HAVE_INVALID_TO_PAY_VALUE,
+        message: 'Cannot pay for Receipt with invalid "toPay" value',
+    },
+    RECEIPT_HAVE_NEGATIVE_TO_PAY_VALUE: {
+        mutation: 'registerMultiPaymentForVirtualReceipt',
+        code: BAD_USER_INPUT,
+        type: RECEIPTS_HAVE_NEGATIVE_TO_PAY_VALUE,
+        message: 'Cannot pay for Receipt with negative "toPay" value',
+    },
+}
+
+const RegisterMultiPaymentForVirtualReceiptService = new GQLCustomSchema('RegisterMultiPaymentForVirtualReceiptService', {
+    types: [
+        {
+            access: true,
+            type: 'enum RegisterMultiPaymentVirtualReceiptCurrency { RUB, USD, KZT, TRY }',
```

Comment:
I meant, couldnt we simply add all currencies? We already have that field -- see BillingIntegration model.. 

We may expect a String for input and check if this string is correct currecny code in business logic

Reply:
We don't have any kind of enumerated currencies in condo. Yes we have a currencyCode in BillingIntegration. But we don't have a BillingIntegration in this service.